openpype/hosts/blender/api/ops.py

```diff
@@ -774,6 +773,52 @@ def execute(self, context):
 
         return {"FINISHED"}
 
+
+class SCENE_OT_DuplicateOpenpypeInstance(bpy.types.Operator):
+    """Duplicate OpenPype instance"""
+
+    bl_idname = "scene.duplicate_openpype_instance"
+    bl_label = "Duplicate OpenPype Instance"
+    bl_options = {"REGISTER", "UNDO"}
+
+    def execute(self, context):
+        op_instances = context.scene.openpype_instances
+
+        # Create new intance
+        new_instance = op_instances.add()
+
+        # Get active instance
+        active_instance = op_instances[
+            context.scene.openpype_instance_active_index
+        ]
+
+        # Apply metadata
+        # NOTE hardcoded because strange behaviour with .items()
+        for k in ["avalon", "creator_name", "icons"]:
+            new_instance[k] = active_instance.get(k)
+
+        # Change name
+        split_name = active_instance.name.rsplit(".", 1)
+        if len(split_name) > 1:
+            basename, number = split_name
+            extension = str(int(number) + 1).zfill(3)
+        else:
+            basename = split_name[0]
+            extension = "001"
+        new_instance.name = f"{basename}.{extension}"
```

Comment:
Numbered extension doesn't work great. We should do this in a while loop with incrementation and check of existing.
![Capture d’écran 2023-02-02 155912](https://user-images.githubusercontent.com/13371155/216362104-331e68c1-0665-457c-8a90-cdd529c89d56.png)

Reply:
I rewrote more, it seems to me more accurate and reliable. You'll tell me.